Dialog/index.d.ts

```diff
@@ -0,0 +1,186 @@
+import MaterialComponent from '../MaterialComponent';
+import { VNode, PreactHTMLAttributes } from 'preact';
+import Button from '../Button';
+import { MDCFoundation, MDCComponent } from '../MaterialComponentsWeb';
+
+declare interface IDialogProps extends JSX.HTMLAttributes {
+  onAccept?: () => void;
+  onCancel?: () => void;
+}
+export default class Dialog extends MaterialComponent<IDialogProps, {}> {
+  static Header: typeof Header;
+  static Body: typeof Body;
+  static Footer: typeof Footer;
+  static FooterButton: typeof FooterButton;
+
+  MDComponent: MDCDialog;
+}
+
+declare class Header extends MaterialComponent<JSX.HTMLAttributes, {}> {}
+
+declare interface IBodyProps extends JSX.HTMLAttributes {
+  scrollable?: boolean;
+}
+declare class Body extends MaterialComponent<IBodyProps, {}> {}
+
+declare class Footer extends MaterialComponent<JSX.HTMLAttributes, {}> {}
+
+declare interface IFooterButtonProps extends HTMLAttributesWithoutAccept {
+  accept?: boolean;
+  cancel?: boolean;
+}
+declare class FooterButton extends Button<IFooterButtonProps, {}> {}
+
+declare class MDCDialogFoundation extends MDCFoundation<MDCDialog> {
+  open(): void;
+  close(): void;
+  isOpen(): boolean;
+  accept(shouldNotify: boolean): void;
+  cancel(shouldNotify: boolean): void;
+}
+
+declare class MDCDialog extends MDCComponent<MDCDialogFoundation> {
+  open(): boolean;
+  show(): void;
+  close(): void;
+}
+
+/**
+ * `IFooterButtonProps` conflicts with `JSX.HTMLAttributes` as
```

Comment:
Maybe i am seeing this wrong, but couldn't we use something like:
http://ideasintosoftware.com/typescript-advanced-tricks/
`Omit<JSX.HTMLAttributes, 'accept'>`

Reply:
That might be the best option we have. @cvuorinen, what do you think?

Reply:
Yes, that looks good. :+1: I was looking for something like that myself but didn't find yet.

Reply:
Cool, i'm happy that it worked.
It's more clean with Omit 👍